components/typography/Heading.js

```diff
@@ -0,0 +1,27 @@
+
+export default function Heading({
+    text,
+    style,
+    weight = 'font-bold',
+    level = '2',
+    textColor = 'text-primary-800',
+    className
+}) {
+    let classNames
+    const Tag = `${level}`;
+
+    if (style === 'h1') {
+        classNames = `text-2xl tracking-heading md:text-6xl leading-snug ${weight || ''} ${className || ''}`
+    } else if (style === 'h2') {
+        classNames = `text-2xl tracking-heading md:text-4xl leading-snug ${weight || ''} ${className || ''}`
+    } else if (style === 'h3') {
+        classNames = `text-2xl tracking-heading md:text-2xl leading-snug ${weight || ''} ${className || ''}`
+    }
+
+    return (
+        <Tag className={`${textColor} ${classNames}`}>
+            {text}
+        </Tag>
+    )
+
+}
```

Comment:
I will describe some tips, because probably it's your first React component :)

If you wanna pass something as child to the component (in your case `text`) it should be treated as children. I know that we are passing text/content etc in other component as props, but we should start moving to the `children` approach. What I am talking about? You can pass content by prop, or by children (content between component) like:

```js
<Heading>
  Some text <- this is React children
</Heading>
```

so I would like to change `text` to the `children` prop - you can check that simple blog post https://www.bigbinary.com/react-best-practices/react-children :)

Also `level` prop should be renamed to the `as` because as I see in some places, you used that component also with paragraph (`<Heading level="p"> ...`), `as` is more verbose and it indicates that you can use also another HTML tags not only headings. And of course `level` default value should be changed to the `h2` not `2` :)

Tbh I also don't like `style` prop's name, because it is similar to the `styles` prop which handles CSS styles. Maybe `styleFor`? 🤔  Also we can remove `weight` and `textColor` props and move default values to the `className` but we have to remember that we should always pass classes for weight and text color... 🤔  We can stay with `weight` and `textColor` in that case.

If you have concerns/question, feel free to raise them! :)

Reply:
"Identifies" for common styles classes/styles like `heading-xl` etc are a great idea! Also about your concerns about `bottlenecks`: we can create such an identifiers and then we can use `className` to override some classes, so it shouldn't be a problem. However identifiers for colors isn't good idea (for that we can use `textColor` prop or `className`).

> I'm not sure that I understand the meaning of as in this context, but if you think it is better I am fine with updating it to that!

`as` means that you wanna render root component of given React component (first HTML tag that occurs in the output of component) "as" given HTML  tag. Maybe for `heading` isn't good but for other case it should be better name of prop:

```js
function Block({ as = 'div' }) {
  const As = as

  return <As ...>
}
```

> This is related to what I was saying above. I agree that style is not a great label. Would typeStyle make more sense here? Or maybe treatment?

Yeah, `typeStyle` is better, that same as `treatment`, but maybe `identifier`? Or `preset`? :)

Reply:
Okay the `as` makes more sense to me now! I think I have a new plan of creating a separate component `Paragraph` for body copy. Then in the future maybe we can create components for lists and other text markup, just to keep the class names a little bit more streamlined! So for `Heading` we can keep the `level` prop as I think it makes more sense if it were only h tags, what do you think?

And I think `typeStyle` might best define what this prop does, as `identifier`, `treatment`, and `preset` might be too vague. What do you think about that?

Reply:
Just added the `Paragraph` component which could resolve the issues you mentioned here. Let me know if that works! 

A question I had as I was going through the code: In some areas, the props on the component are not stacked when being used, but instead they are inline like: `className="text-gray-800" showSupporterBanner="false"` is this treatment okay? Or should we add returns for each prop?

Reply:
> Okay the as makes more sense to me now! I think I have a new plan of creating a separate component Paragraph for body copy. Then in the future maybe we can create components for lists and other text markup, just to keep the class names a little bit more streamlined! So for Heading we can keep the level prop as I think it makes more sense if it were only h tags, what do you think?

Yeah, we should think about UI library and then create a bigger components to render e.g. channels of AsyncAPI spec, single server etc :) We should built them on that simple like `Heading`, `Paragraph` etc.

> And I think typeStyle might best define what this prop does, as identifier, treatment, and preset might be too vague. What do you think about that?

go ahead with `typeStyle` :) 

> Just added the Paragraph component which could resolve the issues you mentioned here. Let me know if that works!

I will look :)

> A question I had as I was going through the code: In some areas, the props on the component are not stacked when being used, but instead they are inline like: className="text-gray-800" showSupporterBanner="false" is this treatment okay? Or should we add returns for each prop?

Sorry, but I don't understand that `...are not stacked when being used...` and `...Or should we add returns for each prop...` 😅  Do you have in mind that some component is written like:

```js
<Component prop1='...' prop2='...'>...</Component> 
```

and it should be as:

```
<Component 
  prop1='...' 
  prop2='...'
>
  ...
</Component> 
```

? Or better will be to paste some link to see that inlined component :)

Reply:
@magicmatatjahu yep exactly like your example, just wondering if there is a "right" way to write out the props. In some places I realized I didn't add returns, like here:

```
<Heading level="h5" typeStyle="heading-md-semibold">
    Join our Slack workspace
</Heading>
```

instead of adding returns:

```
<Heading 
  level="h5"
  typeStyle="heading-md-semibold"
>
    Join our Slack workspace
</Heading>
```

Just wondering which format is preferred.

Reply:
> Yeah, we should think about UI library and then create a bigger components to render e.g. channels of AsyncAPI spec, single server etc :) We should built them on that simple like Heading, Paragraph etc.

Is there an example of this on the website? `channels of AsyncAPI spec` `single server` Or are these things in the Studio HTML template?

Reply:
If you write component like in second case you should remember about indentation, but first and second are fine :)